### PR TITLE
Blog post for enhanced non-OpenAI model support

### PR DESCRIPTION
**Pull request by marklysze**
PR for blog post for the enhanced support of non-OpenAI models through new client classes. Includes Anthropic, Mistral.AI, Together.AI so far.

## Related issue number

See Roadmap issue #2946

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/blog/2024-06-24-AltModels-Classes/index.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+```py
+
+user_proxy.intiate_chat(assistant, message="Write python code to print Hello World!")
+
+```
+
+**NOTE: To integrate this setup into GroupChat, follow the [tutorial](https://microsoft.github.io/autogen/docs/notebooks/agentchat_groupchat) with the same config as above.**
+
+
+## Function Calls
```
Comment (olgavrou) [inline website/blog/2024-06-24-AltModels-Classes/index.mdx:95]: is more text going to be added here? if not they should be removed

**Review by olgavrou — APPROVED** [review]
LGTM when comment is resolved

File: website/blog/2024-06-24-AltModels-Classes/index.mdx (hunk 1 above)
Comment (Hk669) [reply website/blog/2024-06-24-AltModels-Classes/index.mdx:95]: Yes, the function call tutorial will go in.

**Comment by marklysze** [comment]
Thanks @olgavrou, I'll put the function calling in and would like to write some text around using these alternative models including some tips/parameters that are useful to tweak and work with. That will take a day or two to do. Hope that's okay.

**Comment by marklysze** [comment]
I've updated with the upfront text. Still to do is the function calling example.

**Comment by marklysze** [comment]
Thanks @olgavrou and @qingyun-wu, I've added the function calling to the blog post. Have also added a separate notebook for more function calling between different providers (separated it as it was very long).

Let us know what you think and if we can improve it.

File: website/blog/2024-06-24-AltModels-Classes/index.mdx [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+So we can see how Anthropic's Sonnet is able to suggest multiple tools in a single response, with AutoGen executing them both and providing the results back to Sonnet. Sonnet then finishes with a nice email summary that can be the basis for continued real-life conversation with the customer.
+
+## More tips and tricks
+
+For an interesting chess game between Anthropic's Sonnet and Mistral's Mixtral, we've put together a sample notebook that highlights some of the tips and tricks for working with non-OpenAI LLMs. [See the notebook here](https://microsoft.github.io/autogen/docs/notebooks/agentchat_nested_chats_chess_altmodels).
+
+
+### Acknowledgements
+
+This blogpost has been revised with the help of ...
```
Comment (Hk669) [inline website/blog/2024-06-24-AltModels-Classes/index.mdx:392]: @marklysze if there are none, we can remove this section.

**Review by Hk669 — APPROVED** [review]

File: website/blog/2024-06-24-AltModels-Classes/index.mdx (hunk 2 above)
Comment (marklysze) [reply website/blog/2024-06-24-AltModels-Classes/index.mdx:392]: Thanks @Hk669, I left it in there while it gets reviewed and anyone assisting with the review can go in there (as per the last blog), but if no changes required then, yep, we can remove :).

Comment (marklysze) [reply website/blog/2024-06-24-AltModels-Classes/index.mdx:398]: Okay, I've removed for now... will add it in if anyone has edits.

**Comment by marklysze** [comment]
Thanks for approving @olgavrou!

@qingyun-wu, let us know if you would like anyone else to review or if you want any changes. Groq has been added since it is now merged.

**Comment by Hk669** [comment]
@olgavrou @qingyun-wu are we waiting for all the non-OpenAI models to be merged? can you please review this blog and let us know if you would like anyone else to review or if you want any changes.

File: website/blog/2024-06-24-AltModels-Classes/index.mdx (hunk 2 above)
Comment (sonichi) [reply website/blog/2024-06-24-AltModels-Classes/index.mdx:398]: The section is not removed yet.

Comment (marklysze) [reply website/blog/2024-06-24-AltModels-Classes/index.mdx]: Ah, had it ready but didn't fully commit. Done now and I'll bring it up to main as well.

Comment (marklysze) [reply website/blog/2024-06-24-AltModels-Classes/index.mdx]: Thanks, all done.

**Pull request merged**